core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -3117,7 +3117,8 @@ class ReplicaManagerTest {
     isShuttingDown: AtomicBoolean = new AtomicBoolean(false),
     enableRemoteStorage: Boolean = false,
     shouldMockLog: Boolean = false,
-    remoteLogManager: Option[RemoteLogManager] = None
+    remoteLogManager: Option[RemoteLogManager] = None,
+    defaultTopicRemoteLogStorageEnable: Boolean = true
```

Comment:
Default set true is to preserve the same behaviour on other tests. For example, for test `testStopReplicaWithDeletePartitionAndExistingPartitionAndEqualLeaderEpoch`, when `enableRemoteStorage` is set true, it will verify `verify(mockRemoteLogManager).stopPartitions(ArgumentMatchers.eq(Collections.singleton(stopPartition)), any())`, if `defaultTopicRemoteLogStorageEnable ` set false, in this pr change, the topic will be filtered out and test failed in the unit test, which is same as other unit tests. @kamalcph

Reply:
Let's update all the callers instead and supply the flag explicitly:

```
    val replicaManager = setupReplicaManagerWithMockedPurgatories(mockTimer, aliveBrokerIds = Seq(0, 1),
      enableRemoteStorage = enableRemoteStorage, defaultTopicRemoteLogStorageEnable = enableRemoteStorage)
```

The intention is that the default value should resemble the cluster default which is `false`.

Reply:
Went through the test cases, we can keep it as true.